### PR TITLE
reset formats and answers when switching question type

### PR DESCRIPTION
**Pull request by camliu89**
1. Reset the formats and answers when the question type is changed.

### CODE REVIEW

File: shared/specs/model/exercise/question.spec.js [hunk 1]
```diff
@@ -78,9 +76,9 @@ describe('Exercise Question', () => {
         question.collaborator_solution_html = '';
         expect(question.collaborator_solutions).toHaveLength(0);
     });
-    it('uses answers for detecting multi-choice', () => {
+    it('uses formats for detecting multi-choice', () => {
         expect(question.isMultipleChoice).toBe(true);
-        question.answers = [];
+        question.formats = ['free-response'];
         expect(question.isMultipleChoice).toBe(false);
```
Comment (nathanstitt) [inline shared/specs/model/exercise/question.spec.js:82]: we do not want to switch to using formats for determining multiple-choice.  that should continue to rely on if the exercise has any answers, that way it matches what tutor-server does.  cc @Dantemss to confirm the BE still does that though.

This task was about clearing the answers array when the exercise is saved with it's format not set to multiple-choice

File: shared/src/model/exercise/question.js [hunk 2]
```diff
@@ -114,22 +114,32 @@ class ExerciseQuestion extends BaseModel {
...
+          case 'true-false':
+              if (this.answers.length == 0) {
+                  this.answers.push({ content_html: 'True' });
+                  this.answers.push({ content_html: 'False' });
+              }  
+              break;
       }
-      this.formats = uniq(formats).sort();
+
+      uniq(this.formats).sort();
```
Comment (nathanstitt) [inline shared/src/model/exercise/question.js:142]: we do not want modify the answers until the exercise is saved.  Image the case of an editor who types up very long answers, then inadvertently hits the 'free-response' radio input, quickly switches back to multiple-choice and has lost all their work.

**Review by nathanstitt — CHANGES_REQUESTED** [review]
Hey Alex, I think you mis-understood the acceptance criteria on the card:

> Expected behavior:
When you switching to the Open/Ended or True/False option, answers should be deleted.

We only remove answer choices upon saving the question
When user clicks open-ended, we hide the answer choices, but save them in browser memory
When user saves or publishes, we remove answer choices
If user clicks back to multiple-choice option, answer choices will reappear.
Steps to Reproduce the issue:
1. Create an exercise and choose the Multiple Choice option.
2. And some answers.
3. Switch to the Open-Ended or True/False option.
4. Answers are still persisted in the exercise card.

**Comment by camliu89** [comment]
thanks nathan. for some reason, I didn't see the bullet points in the card that says to keep the answers.

**Comment by camliu89** [comment]
Closing this. This change needs a design session to talk about some stuff.

**Pull request closed without merging**